AQCU-1035 CORR- Always show the threshold lane

mhines-usgs opened the pull request:
-creating an empty threshold row with the start/end dates from reportMetadata and a null label; if there are threshold data these are overwritten with actual data, if not, an empty row appears.

mhines-usgs committed:
AQCU-1035 CORR- Always show the threshold lane
-creating an empty threshold row with the start/end dates from reportMetadata and a null label; if there are threshold data these are overwritten with actual data, if not, an empty row appears.

mhines-usgs commented:
![image](https://cloud.githubusercontent.com/assets/6920344/23815141/4d961eae-05ac-11e7-82f5-dab9a79c818c.png)

zmoore-usgs reviewed (changes requested):
These changes seem to work but I think there is a somewhat cleaner way to get the lane to always show up. If you look in the render file the call to `createPlotLanes` takes arguments of `optionalData` and `requiredData`. I think if you move the threshold stuff from optional to required it should make the lane always show. You might have to adjust the math for `rectHeight` in `createPlotLanes` though because I think it's based on the number of required/optional lanes.

mhines-usgs commented:
Thanks @zmoore-usgs -- checking that out. I think I've got it working how you suggest with the few examples from the JIRA now, but the height is still a little funky. Will keep looking.

mhines-usgs committed:
AQCU-1035 updates to logic for always showing thresholds

zmoore-usgs commented:
Woah, my comment ended up in some weird place where it doesn't show up in the files changed view... 😕  I'll try again

zmoore-usgs commented on R/correctionsataglance-render.R:
```diff
@@ -41,7 +41,8 @@ correctionsataglanceReport <- function(reportObject) {
                                   dataLanes[['normalData']][["laneNameYPos"]],
-                                  dataLanes[['postData']][["laneNameYPos"]]))
+                                  dataLanes[['postData']][["laneNameYPos"]],
+                                  dataLanes[['thresholdData']][["laneNameYPos"]]))
```
This part of the code is adjusting the y position of the "processing order" label to be in the middle of the processing order lanes. I'm not sure if threshold data is supposed to go under the "Processing Order" label or not? I don't think it was under it before when the lane show up, though I'm not totally sure.

zmoore-usgs reviewed (approved):
Changes look good, just a question about the processing order label. If threshold data is supposed to be under it then I think this looks good!

mhines-usgs committed:
the threshold data shouldn't be a part of the processing data y-axis label section. derp.

mhines-usgs committed:
fixing test breaking locally

mhines-usgs committed:
fixing tests that changed

mhines-usgs committed:
merge conflict

Merge branch 'master' of github.com:mhines-usgs/repgen

# Conflicts:
#	tests/testthat/test-correctionsataglance.R

mhines-usgs committed:
ok, really fixing tests this time. and unfixing broken fixes i made.
